app/src/main/java/cm/aptoide/pt/promotions/PromotionsService.java

```diff
@@ -101,6 +105,32 @@ public String getWalletAddress() {
     return result;
   }
 
+  public Single<List<PromotionMeta>> getPromotions(String type) {
+    return GetPromotionsRequest.of(type, bodyInterceptorPoolV7, okHttpClient, converterFactory,
+        tokenInvalidator, sharedPreferences)
+        .observe()
+        .map(promotionsResult())
+        .toSingle();
+  }
+
+  @NonNull private Func1<GetPromotionsResponse, List<PromotionMeta>> promotionsResult() {
```

Comment:
Why not creating a method and injecting the result inside of it like we usually do ?

Reply:
In my opinion it makes the getPromotions method cleaner, but I agree that it should be the same pattern we use everywhere else.